### ISSUE TITLE
Using CvlVerb with a table as an impulse

### ISSUE THREAD

**Issue by snuq** [belangeo/pyo#284]
Hello, Ive been working on a composer program that is using pyo as the sound generator backend.  

Im not completely sure if this is a support request or a feature request:
Right now i am attempting to make use of the CvlVerb class, but everything is running in memory (using files extracted from zip files), so i cant reference a fie name as the impulse.  Is there any way to directly assign a table as the impulse for CvlVerb?

**Comment by alexdrymonitis** [comment]
Aparently not. You should probably store your data into a file before loading it to CvlVerb. Python should be capable of providing the necessary tools for that.

**Comment by snuq** [comment]
I tried to dig into the code to troubleshoot this... im just putting this here in case anyone gets the idea to implement this (unfortunately, i dont know any c, or i might try...).

The main issue seems to be that in "pyo/src/objects/fftmodule.c" the "CvlVerb_analyse_impulse" function is hard coded to use sf_open.  If this function was split into "CvlVerb_load_impulse" and "CvlVerb_analyse_impulse", 
the sound data could maybe be passed in from another source.

Alternately, maybe the sf_open function (i cant figure out where in the code this function is actually defined!) could be updated to also accept a bytesio object instead of just a path.
This would be extremely helpful for another area of my project as well, as it seems that pyo has the ability to load various file types but only directly from disk, so i am needing to use another library for loading files.

**Comment by alexdrymonitis** [comment]
I don't think that you need to go down to the C level of Pyo and change things there. Can't you include an algorithm in your script that unzips your files and chooses an audio file that you can save on your drive and then load to SfPlayer (https://belangeo.github.io/pyo/api/classes/players.html#sfplayer)?

**Comment by snuq** [comment]
the way my app is set up, effects can all be used in a per-note or per-channel basis.  having to use this workaround would make this entirely unusable in the first space as it would create a ton of very slow unnecessary file access.  
maybe I can come up with some way to cache/reuse CvlVerb instances, but ive also been looking into trying to figure out how to make my own PyoObject that will just let me run the input buffer through my own processing through numpy/scipy
